llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -26193,9 +26193,21 @@ static SDValue performFlagSettingCombine(SDNode *N,
   }
 
   // Combine identical generic nodes into this node, re-using the result.
-  if (SDNode *Generic = DCI.DAG.getNodeIfExists(
-          GenericOpcode, DCI.DAG.getVTList(VT), {LHS, RHS}))
-    DCI.CombineTo(Generic, SDValue(N, 0));
+  auto CombineWithExistingGeneric = [&](SDValue Op0, SDValue Op1) {
```

Comment:
nit: I found "`With`" a little confusing here (as the generic operation is not being combined). 
```suggestion
  auto CombineToExistingGeneric = [&](SDValue Op0, SDValue Op1) {
```

Also, the comment above would be outdated in the commuted operands case (as the existing generic node is not _identical_, but equivalent) .